dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
This conversation should probably move into a designated issue, but I would throw this idea out there:

```
VisualAddon
{
  Eigen::Vector4d mDefaultColor;
  std::shared_ptr< std::vector<Eigen::Vector4d> > mVertexColors;
  std::shared_ptr< TextureData > mTexture;

  enum ColorMode {
    DEFAULT_COLOR = 0,
    PER_VERTEX,
    TEXTURE
  };

  ColorMode mColorMode;

  /* ... Other VisualAddon stuff ... */
};
```

This way color and geometry data can be decoupled. If ``mVertexColors`` or ``mTexture`` being nullptr will imply that there is no per-vertex or texture coloring available. If ``mColorMode`` is set to an unavailable color mode, then it will simply use ``mDefaultColor``. If ``mVertexColors`` or ``mTexture`` is incompatible with the ``MeshShape`` of the ``ShapeNode``, then the renderer should print a message (once) that informs the user of this incompatibility and then use ``mDefaultColor`` instead.

I think this approach would satisfy everyone's goals while also being extremely memory efficient. The catch would just be the effort of implementation.